packages/syft/src/syft/core/tensor/smpc/mpc_tensor.py

```diff
@@ -170,11 +230,39 @@ def _get_shares_from_local_secret(
 
         return shares
 
-    def reconstruct(self) -> Any:
+    def request(
+        self,
+        reason: str = "",
+        block: bool = False,
+        timeout_secs: Optional[int] = None,
+        verbose: bool = False,
+    ) -> None:
+        for child in self.child:
+            child.request(
+                reason=reason, block=block, timeout_secs=timeout_secs, verbose=verbose
+            )
+
+    def reconstruct(self):
         # TODO: It might be that the resulted shares (if we run any computation) might
         # not be available at this point
 
-        local_shares = [share.get_copy() for share in self.child]
+        # for now we need to convert the values coming back to int32
+        # sometimes they are floats coming from DP
+        def convert_child_numpy_type(tensor: Any, np_type: type) -> Any:
+            if isinstance(tensor, np.ndarray):
+                return np.array(tensor, np_type)
+            if hasattr(tensor, "child"):
+                tensor.child = convert_child_numpy_type(
+                    tensor=tensor.child, np_type=np_type
+                )
+            return tensor
+
+        local_shares = []
+        for share in self.child:
+            res = share.get()
+            res = convert_child_numpy_type(res, np.int32)  # TODO: change to np.int64
```

Comment:
Add np.int64 / precision tensor handling plus tests for these overflow issues to backlog.

Reply:
I'm eliminating for now as we're exclusively focusing on np.int32 support and will tackle int64 separately.